staging/src/k8s.io/sample-controller/pkg/client/informers/externalversions/samplecontroller/v1alpha1/foo.go

```diff
@@ -39,18 +39,29 @@ type FooInformer interface {
 
 type fooInformer struct {
 	factory internalinterfaces.SharedInformerFactory
+	filter  internalinterfaces.FilterFunc
 }
 
 // NewFooInformer constructs a new informer for Foo type.
 // Always prefer using an informer factory to get a shared informer instead of getting an independent
 // one. This reduces memory footprint and number of connections to the server.
 func NewFooInformer(client versioned.Interface, namespace string, resyncPeriod time.Duration, indexers cache.Indexers) cache.SharedIndexInformer {
```

Comment:
so we had a namespace argument already? What does this PR add then (other than the posibility to tweak the ListOptions) ?

Reply:
Yep we've always had a `namespace` argument on the `NewFooInformer` (or 'leaf' constructors as I like to think of them).

The issue is a few lines below - the `Informer` function (which forms part of the SharedInformerFactory) hardcodes `NamespaceAll`: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/sample-controller/pkg/client/informers/externalversions/samplecontroller/v1alpha1/foo.go#L64

This means that in order to construct namespaced informers, you cannot use the SharedInformerFactory, which is what the original issue was about 😄 . In order to get around this in the past, I've created a SharedInformerFactory implementation that is non-type aware, and instead just returns SharedIndexInformer instances (I then use a GVK structure as a 'key' for the 'informer caching'. This gets super ugly, and really difficult to keep track of/inconsistent)